app/controllers/login_controller.rb

```diff
@@ -7,11 +7,15 @@ def logout
   def public
     request.env['warden'].logout if github_authenticated? :private
     github_authenticate! :default
+    @user = gh.user
+    @emails = gh.user.emails
```

Comment:
this is a nuance of the design of ghee 

you are not forcing an API call immediate and thus there is a risk an exception in the after_action callback

this code is better written as 

```
@user = gh.user
@emails = gh.users(@user[:login]).emails
```